### ISSUE TITLE
Multiple execution on plugin lose defined properties

### ISSUE THREAD

**Issue by mhermosi** [fabric8io/docker-maven-plugin#1226]
I foud this when testing parallel build, but then on single build appeared.

Before I had the plugin defined twice instead of using executions.

when switched to executions started to happen.

Versions and execution command:

```
Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T14:33:14-04:00)
Maven home: /Users/user/Development/maven
Java version: 12.0.1, vendor: Oracle Corporation, 
runtime: /Library/Java/JavaVirtualMachines/openjdk-12.0.1.jdk/Contents/Home

command:
mvn -B -s ./settings.xml -Dmaven.test.skip=true clean install docker:build docker:push -P private-registry -T 1C


```

I have another thing about parallel build. When using -T option on maven and on single or multiple executions the plugin lose the defined properties:

- properties:

`${container.registry}, ${image.prefix} and ${project.version}`

- pom.xml extract:

```xml
<plugin>
   <groupId>io.fabric8</groupId>
      <artifactId>docker-maven-plugin</artifactId>
      <executions>
            <execution>
                  <id>demo-ui-build</id>
                  <configuration>
                        <registry>${container.registry}</registry>
                        <images>
                           <image>
                              <name>${image.prefix}/${project.artifactId}:${project.version}</name>
                              <build>
                                 <dockerFileDir>${project.basedir}</dockerFileDir>
                                 <dockerFile>Dockerfile</dockerFile>
                              </build>
                           </image>
                     </images>
                  </configuration>
            </execution>
            <execution>
                  <id>app-ui-build</id>
                  <configuration>
                        <registry>${container.registry}</registry>
                        <images>
                           <image>
                              <name>${image.prefix}/app-ui:${project.version}</name>
                              <build>
                                 <dockerFileDir>${project.basedir}</dockerFileDir>
                                 <dockerFile>Dockerfile</dockerFile>
                              </build>
                           </image>
                     </images>
                  </configuration>
            </execution>
         </executions>
  </plugin>
```

For some reason `${project.artifactId}` is not lost.... but `${project.version}` is lost and seems to be defaulted to `latest`

The weird thing is that `${image.prefix}` seems to be getting the last part of the parent pom.xml groupId I mean.. if artifact is something like `com.domain.xxx.yyy:artifactName:0.0.1-SNAPSHOT` and `${image.prefix} = base` then the result of image name is `yyy/artifactName:latest` instead of `base/artifactName:0.0.1-SNAPSHOT`

`${container.registry}` is being set on the profile definition.

In conclusion, I think this plugin should review its coding to support thread-safety so we can use the whole potential of

_Originally posted by @mhermosi in https://github.com/fabric8io/docker-maven-plugin/issues/905#issuecomment-493177981_

**Comment by mhermosi** [comment]
if I use these definitions inside execution when running a single execution in the plugin and single maven build (no -T XX) docker:build is fine but docker:push fails with the same problem

´´´xml
<phase>package</phase>
<goals>
   <goal>build</goal>
   <goal>push</goal>
</goals>
```

**Comment by mhermosi** [comment]
Found the issue.

I was executing maven-exec-plugin on package phase and also docker plugin on package phase. I moved maven-exec-plugin to compile (because I am doing some javascript build) and docker to install and seems to be working fine ... even for parallel build

**Comment by mhermosi** [comment]
Found the issue.

I was executing maven-exec-plugin on package phase and also docker plugin on package phase. I moved maven-exec-plugin to compile (because I am doing some javascript build) and docker to install and seems to be working fine ... even for parallel build
